runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClusterState.java

```diff
@@ -69,4 +69,22 @@ public boolean isReady(){
 
         return true;
     }
+
+    /**
+     * See if cluster is ready. If so, check if the cluster has at least 1 primary sequencer that's READY.
+     *
+     * @return if primary sequencer is ready
+     */
+    public boolean isPrimarySequencerReady() {
+        if (nodes.isEmpty()) {
+            return false;
+        }
+        for (NodeState nodeState : nodes.values()) {
+            if (nodeState.getSequencerMetrics().getSequencerStatus() == SequencerMetrics.SequencerStatus.READY) {
```

Comment:
At any given point in time, there can only be 1 primary sequencer. You need to pass the endpoint of the primary sequencer currently active in the current layout and only check it's status. The return type of this method should be SequencerStatus corresponding to the status of the primary sequencer.

Reply:
What is the best way to find out the endpoint of the primary sequencer from the nodes?

Reply:
You can find it from the layout.
`Layout.getPrimarySequencer`

Reply:
Thank you!